scarlet2/scene.py

```diff
@@ -147,7 +147,7 @@ def update_fn(updates, state, params):
                 if params is None:
                     raise ValueError(base.NO_PARAMS_MSG)
                 updates = jax.tree_util.tree_map(
-                    lambda u, step, param: -step * u if not callable(step) else -step(param) * u,
+                    lambda u, step, param: -step * u if not callable(step) else -step(param,niter) * u,
```

Comment:
I'm surprised that this works because `niter` is only defined afterwards. Also, I don't see the corresponding change to `relative_step`, which allows the iteration counter to be passed as 2nd argument.

Reply:
So, let's move the declaration of `niter` above `def scale_by_stepsize()`, call it `it = 0`, and avoid the apparent shadowing of `step` by writing this as
```
lambda u, s, p: -s * u if not callable(s) else -s(p, it) * u,
```